### PR TITLE
Qualify columns with table name in generated SQL

### LINKED ISSUE #4129: Ambiguous column error with chained search
Search: `ProjectMembership?user:User.email=xyz@example.com`

Error:

```json
{
  "outcome": {
    "resourceType": "OperationOutcome",
    "issue": [
      {
        "severity": "error",
        "code": "invalid",
        "details": {
          "text": "column reference \"project\" is ambiguous"
        }
      }
    ]
  }
}
```

image

### PR DESCRIPTION
**Pull request by mattwiller**
Fixes #4129

### CODE REVIEW

File: packages/server/src/fhir/search.ts [hunk 1]
```diff
@@ -733,27 +739,38 @@ function buildFilterParameterComparison(
...
  * @returns The select query condition.
  */
-function buildStringSearchFilter(details: SearchParameterDetails, operator: Operator, values: string[]): Expression {
+function buildStringSearchFilter(
+  table: string,
+  details: SearchParameterDetails,
+  operator: Operator,
+  values: string[]
+): Expression {
+  const column = new Column(details.array ? undefined : table, details.columnName);
```
Comment (codyebberson) [inline packages/server/src/fhir/search.ts:754]: `undefined`?  Curious about why that was necessary.

File: packages/server/src/fhir/search.ts [hunk 2]
```diff
@@ -627,21 +627,27 @@ function buildNormalSearchFilterExpression(
...
+      new Column(table, details.columnName),
+      fhirOperatorToSqlOperator(filter.operator),
+      filter.value
+    );
   } else {
     const values = filter.value
       .split(',')
-      .map((v) => new Condition(details.columnName, fhirOperatorToSqlOperator(filter.operator), v));
+      .map(
+        (v) => new Condition(new Column(undefined, details.columnName), fhirOperatorToSqlOperator(filter.operator), v)
```
Comment (codyebberson) [inline packages/server/src/fhir/search.ts:649]: `undefined`?

**Review by codyebberson — APPROVED** [review]
lgtm 👍 

2 questions on `undefined` table name.  I'm sure there was good reasons, just not obvious to me.

File: packages/server/src/fhir/search.ts (hunk 1 above)
Comment (mattwiller) [reply packages/server/src/fhir/search.ts:754]: This is basically a workaround carried forward from the existing implementation, which requires it to deal with the `ArraySubquery` where the inner subquery needs to refer to the unqualified column name (aliased from `unnest()`)

**Pull request merged**